index.js

```diff
@@ -11,3 +10,52 @@ const mongoose = require('./lib/');
 module.exports = mongoose;
 module.exports.default = mongoose;
 module.exports.mongoose = mongoose;
+
+// Re-export for ESM support
+module.exports.cast = mongoose.cast;
+module.exports.STATES = mongoose.STATES;
+module.exports.setDriver = mongoose.setDriver;
+module.exports.set = mongoose.set;
+module.exports.get = mongoose.get;
+module.exports.createConnection = mongoose.createConnection;
+module.exports.connect = mongoose.connect;
+module.exports.disconnect = mongoose.disconnect;
+module.exports.startSession = mongoose.startSession;
+module.exports.pluralize = mongoose.pluralize;
+module.exports.model = mongoose.model;
+module.exports.deleteModel = mongoose.deleteModel;
+module.exports.modelNames = mongoose.modelNames;
+module.exports.plugin = mongoose.plugin;
+// module.exports.connection = mongoose.connection;
```

Comment:
why not export the default connection, is it also because of "custom driver may overwrite this"?

Reply:
Yes exactly. `setDriver()` can recreate the default connection.